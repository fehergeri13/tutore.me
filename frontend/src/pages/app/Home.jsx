import React from 'react';
import {observable, toJS} from 'mobx';
import {observer, inject} from 'mobx-react';
import "./home.less"
import PostItem from "../../components/post/PostItem";
import Modal from 'react-modal';

@inject('model')
@observer
export default class Home extends React.Component {

    @observable.shallow allPosts = [];

    async componentDidMount() {
        const response = await this.props.model.rest.getPosts();
        this.allPosts.replace(response.posts);

        console.log('this.allPosts', toJS(this.allPosts));
    }

    renderPosts() {
        if (this.allPosts.length === 0) {
            return <h2>Nincs hirdetés</h2>
        }

        return <ul>
            {this.allPosts.map(post => <li key={post.id}><PostItem post={post} isControl={false}/></li>)}
        </ul>;
    }

    render() {
        return <div className="home-page">
            <div className="search">
                <input type="text" id="search-text" placeholder="keresés"/>

                <fieldset>
                    <label><input type="radio" name="contact" value="keres"/>Keres</label>
                    <label><input type="radio" name="contact" value="keres"/>Kínál</label>
                </fieldset>

                <fieldset>
                    <legend>
                        Tantárgy
                    </legend>
                    <label><input type="checkbox" name="contact" value="keres"/>Informatika</label>
                    <label><input type="checkbox" name="contact" value="keres"/>Matematika</label>
                    <label><input type="checkbox" name="contact" value="keres"/>Fizika</label>
                    <label><input type="checkbox" name="contact" value="keres"/>Kémia</label>
                </fieldset>
            </div>

            <div className="content">
                {this.renderPosts()}
            </div>
        </div>;
    }
}
