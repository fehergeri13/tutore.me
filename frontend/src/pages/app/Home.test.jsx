import React from 'react';
import renderer from 'react-test-renderer';
import Home from "./Home";
import {Provider} from "mobx-react";

describe('Home', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <Provider model={{}}><Home /></Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
