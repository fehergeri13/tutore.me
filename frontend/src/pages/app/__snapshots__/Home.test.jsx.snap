// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="home-page"
>
  <div
    className="search"
  >
    <input
      id="search-text"
      placeholder="keresés"
      type="text"
    />
    <fieldset>
      <label>
        <input
          name="contact"
          type="radio"
          value="keres"
        />
        Keres
      </label>
      <label>
        <input
          name="contact"
          type="radio"
          value="keres"
        />
        Kínál
      </label>
    </fieldset>
    <fieldset>
      <legend>
        Tantárgy
      </legend>
      <label>
        <input
          name="contact"
          type="checkbox"
          value="keres"
        />
        Informatika
      </label>
      <label>
        <input
          name="contact"
          type="checkbox"
          value="keres"
        />
        Matematika
      </label>
      <label>
        <input
          name="contact"
          type="checkbox"
          value="keres"
        />
        Fizika
      </label>
      <label>
        <input
          name="contact"
          type="checkbox"
          value="keres"
        />
        Kémia
      </label>
    </fieldset>
  </div>
  <div
    className="content"
  >
    <ul>
      <li>
        <h2>
          <a
            href="#"
          >
            Matematika korrepetálás
          </a>
        </h2>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laudantium nisi perspiciatis voluptas. A aut consectetur deleniti, ea impedit laboriosam minima molestias non numquam, placeat tempore temporibus ullam. Itaque, provident quae?
        </p>
        <div
          className="time"
        >
          2017. 11. 08. 23:16
        </div>
      </li>
      <li>
        <h2>
          <a
            href="#"
          >
            Matematika korrepetálás
          </a>
        </h2>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laudantium nisi perspiciatis voluptas. A aut consectetur deleniti, ea impedit laboriosam minima molestias non numquam, placeat tempore temporibus ullam. Itaque, provident quae?
        </p>
        <div
          className="time"
        >
          2017. 11. 08. 23:16
        </div>
      </li>
      <li>
        <h2>
          <a
            href="#"
          >
            Matematika korrepetálás
          </a>
        </h2>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laudantium nisi perspiciatis voluptas. A aut consectetur deleniti, ea impedit laboriosam minima molestias non numquam, placeat tempore temporibus ullam. Itaque, provident quae?
        </p>
        <div
          className="time"
        >
          2017. 11. 08. 23:16
        </div>
      </li>
      <li>
        <h2>
          <a
            href="#"
          >
            Matematika korrepetálás
          </a>
        </h2>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laudantium nisi perspiciatis voluptas. A aut consectetur deleniti, ea impedit laboriosam minima molestias non numquam, placeat tempore temporibus ullam. Itaque, provident quae?
        </p>
        <div
          className="time"
        >
          2017. 11. 08. 23:16
        </div>
      </li>
      <li>
        <h2>
          <a
            href="#"
          >
            Matematika korrepetálás
          </a>
        </h2>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laudantium nisi perspiciatis voluptas. A aut consectetur deleniti, ea impedit laboriosam minima molestias non numquam, placeat tempore temporibus ullam. Itaque, provident quae?
        </p>
        <div
          className="time"
        >
          2017. 11. 08. 23:16
        </div>
      </li>
    </ul>
  </div>
</div>
`;
